hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/troubleshooting_s3a.md

```diff
@@ -547,6 +547,20 @@ When trying to write or read SEE-KMS-encrypted data, the client gets a
 The caller does not have the permissions to access
 the key with which the data was encrypted.
 
+### <a name="access_denied_requester_pays"></a>`AccessDeniedException` when using a "Requester Pays" enabled bucket
+
+When making cross-account requests to a requester pays enabled bucket, all calls must acknowledge via a header that the requester will be billed.
+
+If you don't enable this acknowledgement within S3A, then you will see a message similar to this:
+
+```
+java.nio.file.AccessDeniedException: s3a://my-bucket/my-object: getFileStatus on s3a://my-bucket/my-object:
```

Comment:
This seems like a generic AccessDeniedException. Is there any way caller will know that this is a requester pays bucket and I have to enable the flag? Otherwise debugging this will be a bit harder no?

Reply:
Unfortunately, "403 Forbidden" is the only information given.

This will only appear if the header is not provided and the call is made by a principal who is not the bucket owner. The bucket owner does not have to provide the header.

Below is HTTP output from the AWS CLI when making a failed GetObject.

```
2022-02-28 14:35:11,318 - MainThread - urllib3.connectionpool - DEBUG - https://<MY-BUCKET>.s3.eu-west-1.amazonaws.com:443 "GET /file.txt HTTP/1.1" 403 None
2022-02-28 14:35:11,319 - MainThread - botocore.parsers - DEBUG - Response headers: {'x-amz-request-id': 'NC3P56B2VZ87PMKF', 'x-amz-id-2': '<TRUNCATED>', 'Content-Type': 'application/xml', 'Transfer-Encoding': 'chunked', 'Date': 'Mon, 28 Feb 2022 14:35:11 GMT', 'Server': 'AmazonS3'}
2022-02-28 14:35:11,320 - MainThread - botocore.parsers - DEBUG - Response body:
b'<?xml version="1.0" encoding="UTF-8"?>\n<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>NC3P56B2VZ87PMKF</RequestId></Error>'
```

Reply:
Okay can't do much here then.

Reply:
in the troubleshooting doc, add a link to this
https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/troubleshooting_s3a.md#access_denied

Reply:
Sorry, I didn't understand @steveloughran.

Did you mean to add a link from here to the [explanation of S3 Requester Pays feature in the index](https://github.com/dannycjones/hadoop/blob/4abf17d729a179b9fd787378de493daff05a3707/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/index.md#requester-pays-buckets)?

Reply:
we have a troubleshooting doc, with a section on possible causes of an access denied (can't acces file;/bucket, can't get at KMS key etc). adding one more "no requester pays flag on a pay to use bucket

Reply:
That's this file/change, no?

Reply:
yes, maybe i was the confused one. sorry , i must have thought it was elsewhere